prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -576,12 +569,11 @@ class DetailedSubmissionHistory(
             }
         }
 
-        if (finishedDestinations >= realDestinations.size) {
+        if (finishedDestinations >= destinations.size) {
             return Status.DELIVERED
-        } else if (finishedDestinations > 0) {
+        } else if (finishedDestinations >= realDestinations.size) {
```

Comment:
Yeah, at that point we're sure that we didn't get stuff delivered to every destination (so not `Delivered`). The reason we switched from `> 0` to `realDestinations` is so that we're sure that at least one destination has been completed before saying that anything has been delivered.
This is an extremely rare distinction that I haven't actually seen happen, but in case down the road we get heavy slowdown during the delivery process this would be more accurate.

Reply:
Let's comment this in the code

Reply:
Added a couple comments that should help